core/meta/object-descriptor.js

```diff
@@ -196,31 +197,31 @@ ObjectDescriptor.addClassProperties(
                         ? this._eventDescriptors.push.apply(this._eventDescriptors, value)
                         : (this._eventDescriptors = value);
                 }
-                value = deserializer.getProperty("propertyValidationRules");
+                value = deserializer.getProperty("validationRules");
                 if (value) {
-                    let _propertyValidationRules = this._propertyValidationRules;
-                    if (_propertyValidationRules && Object.keys(_propertyValidationRules) > 0) {
+                    let _validationRules = this._validationRules;
+                    if (_validationRules && Object.keys(_validationRules) > 0) {
                         for (
                             let i = 0, keys = Object.keys(value), countI = keys.length, iKey, iValue, oValue;
                             i < countI;
                             i++
                         ) {
                             iValue = value[(iKey = keys[i])];
-                            oValue = _propertyValidationRules[iKey];
+                            oValue = _validationRules[iKey];
                             if (iValue && oValue) {
                                 //Merge
                                 if (iValue instanceof Array) {
-                                    _propertyValidationRules.push.apply(_propertyValidationRules, iValue);
+                                    _validationRules.push.apply(_validationRules, iValue);
                                 } else {
                                     //Overwrite
-                                    _propertyValidationRules[iKey] = iValue;
+                                    _validationRules[iKey] = iValue;
```

Comment:
In this case I think we'd want to create an array to contain both, not have one replace the other? If iValue and oValue aren't arrays?